task_script_utils/datetime_parser/datetime_info.py

```diff
@@ -0,0 +1,807 @@
+            tokens (List): a list containing two
+            numeric tokens
+
+        Raises:
+            AmbiguousDateError: When we fail to decide between day and month
+
+
+        Returns:
+            tuple: (day, month)
+        """
+
+        first_token_is_month = int(tokens[0]) <= 12
+        second_token_is_month = int(tokens[1]) <= 12
+
+        if first_token_is_month and second_token_is_month:
+            raise AmbiguousDateError(
+                f"Can't decide day and month between: {tokens}"
+            )
+
+        if not first_token_is_month and not second_token_is_month:
+            # Both tokens could be month
+            raise AmbiguousDateError(
+                f"Can't decide day and month between: {tokens}"
+            )
+
+        day, month = (
+            (tokens[0], tokens[1])
+            if second_token_is_month
+            else (tokens[1], tokens[0])
+        )
+        return day, month
+
+    def _pad_and_validate_offset_value(self, offset):
+        if ":" in offset:
+            # 5:30 --> 05:30
+            # 05:30 --> 05:30
+            if len(offset) == 4:
+                return f"0{offset}"
+            elif len(offset) == 5:
+                return offset
+            else:
+                raise InvalidOffsetError(offset)
+        else:
+            # 2 --> 02:00 | 09 --> 09:00 | 12 --> 12:00
+            # 530 --> 05:30
+            # 0930 --> 09:30 | 1200 --> 12:00
+            if len(offset) == 1:
+                return f"0{offset}:00"
+            elif len(offset) == 2:
+                return f"{offset}:00"
+            elif len(offset) == 3:
+                return f"0{offset[0]}:{offset[1:]}"
+            elif len(offset) == 4:
+                return f"{offset[:2]}:{offset[2:]}"
+            else:
+                raise InvalidOffsetError(offset)
+
+    def _replace_single_characters(self, string: str):
+        # Input =  2018-13-09T11:12:23.000-05:30
+        # output = 2018-13-09 11:12:23.000-05:30
+        char_list = list(string)
+        for i, current_char in enumerate(char_list[1:-1]):
+            if (
+                char_list[i-1].isdigit()
+                and char_list[i].isalpha()
+                and char_list[i+1].isdigit()
+            ):
+                char_list[i] = " "
+
+        return "".join(char_list)
+
+    def _is_format(self, date_str: str, format: str):
+        """Given a date string and a format,
+        try to parse the date.
+
+        Args:
+            date_str (str): date string
+            format (str): date format built using
+            tokens used in `pendulum` library
+
+        Returns:
+            tuple: If date is parsed successfully return
+            (True, datetime object) else return (False, None)
+        """
+        try:
+            parsed = pendulum.from_format(date_str, format)
+            return parsed
+        except Exception as e:
+            return None
+
+    def _try_formats(self, date_str: str, formats: Tuple[str]):
```

Comment:
[minor] Just so the type checker knows it's ok to have more than one string in the tuple, although I think you don't need to require a `Tuple`, but rather a more generic type (`Iterable`?)

```suggestion
    def _try_formats(self, date_str: str, formats: Tuple[str, ...]):
```